### PR TITLE
Update to use the new dat-worker-pool

### PR DESCRIPTION
**Pull request by jcredding**
This updates qs to use the new dat-worker-pool. dat-worker-pool's
interface has changed but functionally it works the same as it
previously did.

A daemon can now be configured with a worker class instead of
worker procs. This is because dat-worker-pool now requires a worker
class when its initialized. A block is no longer part of
dat-worker-pool's interface. The logic for processing a work item
(a queue item for qs) is now part of the worker class that is passed
in. Worker callbacks should now be added to the worker class thats
configured.

The daemon now also builds a worker pool when its initialized
instead of waiting till its started. This was previously made
possible when dat-worker-pool was changed to not start
automatically but qs didn't take advantage of it til now. This
avoids some conditionals and having to set/unset the worker pool
ivar as the daemon is started/stopped.

As part of this change, qs now also provides a `Worker` mixin which
as mentioned before, contains the logic for processing a queue
item. It also contains the on-available and on-error logic that
was previously buried in the `Daemon` class. This mixin is
required for any worker class that is configured on a daemon. The
worker mixin also protects itself from its included hook being run
multiple times to avoid on-error and on-available callbacks being
added multiple times.

This also includes a number of small changes related to using the
new dat-worker-pool:

* Min and max number of workers are no longer configurable because
  dat-worker-pool no longer supports a min or max. A constant
  number of workers is all that can be configured. It now defaults
  to 4 workers (its previous max value).
* Changed the work loop organization to try and make it more
  readable and consistent with dat-tcp/dat-worker-pool.
* Updated styles and conventions to our latest standards
  throughout.

@kellyredding - Ready for review. The benchmarks are slower but I'm pretty sure its just my machine's current state, I'm getting the same speeds on master currently. I'll try to get a better benchmark or just reset the current ones in master with a hotfix. I'm not sure if I've changed OS version since running the benchmarks last.

### CODE REVIEW

File: lib/qs/worker.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      private
+
+      # * This only catches errors that happen outside of running the payload
+      #   handler. The only known use-case for this is dat worker pools
+      #   hard-shutdown errors.
+      # * If there isn't a queue item (this can happen when an idle worker is
+      #   being forced to exit) then we don't need to do anything.
+      # * If we never started processing the queue item, its safe to requeue it.
+      #   Otherwise it happened while processing so the payload handler caught
+      #   it or it happened after the payload handler which we don't care about.
```
Comment (kellyredding) [inline lib/qs/worker.rb]: @jcredding we've gone away from this "bulleted" comment style - can we clean this up?

**Comment by kellyredding** [comment]
@jcredding there are a few other places in the code base where we are doing "bulleted" comments - can we remove those bullets and cleanup the comments throughout?  (I'm fine doing this in a hotfix too).

**Comment by kellyredding** [comment]
@jcredding a couple of minor comments (that can be handled in a sep hotfix.  Otherwise, :-1:

**Comment by jcredding** [comment]
@kellyredding - I'll get the other comments with bullets in a separate effort or hotfix them as needed. I like to try and keep things somewhat focused and some of the comments are in totally unrelated files.

**Pull request merged**